src/game/logic/object/weapon.cpp

```diff
@@ -954,6 +1520,216 @@ void WeaponTemplate::Deal_Damage_Internal(ObjectID source_id,
+                    return;
+                }
+
+                bool suicide = false;
+                DamageInfo info;
+                float angle;
+                Coord3D source_to_target_dir;
+
+                if (source != nullptr && target != victim) {
+                    if ((affects_mask & AFFECTS_SUICIDE) != 0 && source == target) {
+                        suicide = true;
+                    } else {
+                        if ((affects_mask & AFFECTS_SELF) == 0) {
+                            if (source == target) {
+                                goto next;
+                            }
+
+                            if (source->Get_Producer_ID() == target->Get_ID()) {
+                                goto next;
+                            }
+                        }
+
+                        if ((affects_mask & AFFECTS_NOT_SIMILAR) != 0) {
+                            if (source->Get_Template()->Is_Equivalent_To(target->Get_Template())) {
+                                if (source->Get_Relationship(target) == ALLIES) {
+                                    goto next;
+                                }
+                            }
+                        }
+
+                        if ((affects_mask & AFFECTS_NOT_AIRBORNE) != 0 && target->Is_Significantly_Above_Terrain()) {
+                            goto next;
+                        }
+
+                        Relationship relationship = source->Get_Relationship(target);
+                        WeaponAffectsType affects;
+
+                        if (relationship == ALLIES) {
+                            affects = AFFECTS_ALLIES;
+                        } else if (relationship == ENEMIES) {
+                            affects = AFFECTS_ENEMIES;
+                        } else {
+                            affects = AFFECTS_NEUTRALS;
+                        }
+
+                        if ((affects & affects_mask) == 0) {
+                            goto next;
+                        }
+                    }
+                }
+
+                info.m_in.m_damageType = damage;
+                info.m_in.m_deathType = death;
+                info.m_in.m_sourceID = source_id;
+                info.m_in.m_playerMask = 0;
+                info.m_in.m_damageStatusType = status;
+                source_to_target_dir.Zero();
+
+                if (target != nullptr && source != nullptr) {
+                    source_to_target_dir.Set(target->Get_Position());
+                    source_to_target_dir.Sub(source->Get_Position());
+                }
+
+                angle = Get_Radius_Damage_Angle();
+
+                if ((angle >= DEG_TO_RADF(180.0f) || target != nullptr) && source != nullptr) {
+                    Vector3 source_forward_dir;
+                    source->Get_Transform_Matrix()->Get_X_Vector(&source_forward_dir);
+                    Vector3 v2(source_to_target_dir.x, source_to_target_dir.y, source_to_target_dir.z);
```

Comment:
Lets do

`source_to_target_dir` => `source_to_target_vec`
`v2` => `source_to_target_dir`

Then it should be clear I think.

Reply:
fixed.